File: number_of_subsequences_that_satisfy_the_given_sum_condition.java
class Solution {
    public int numSubseq(int[] nums, int target) {
        Arrays.sort(nums);
        int[] pow = new int[nums.length];
        pow[0] = 1;

        for (int i = 1; i < nums.length; i++) {
            pow[i] = (pow[i - 1] * 2) % 1000000007;
        }

        int left = 0, right = nums.length - 1;
        int res = 0;

        while (left <= right) {
            if (nums[left] + nums[right] > target) {
                right--;
            } else {
                res = (res + pow[right - left]) % 1000000007;
                left++;
            }
        }

        return res;    
    }
}

File: number_of_subsequences_that_satisfy_the_given_sum_condition.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var numSubseq = function(nums, target) {
    let pow = [];
    pow.push(1);

    nums.sort((a, b) => { return a - b; })

    const mod = 1e9+7;

    for (let i = 1; i < nums.length; i++) {
        pow.push((pow[pow.length - 1] * 2) % mod);
    }

    let left = 0, right = nums.length - 1, res = 0;

    while (left <= right) {
        if (nums[left] + nums[right] > target) {
            right--;
        } else {
            res = (res + pow[right - left]) % mod;
            left++;
        }
    }

    return res;
};

